include/chaiscript/chaiscript_threading.hpp

```diff
@@ -107,7 +107,7 @@ namespace chaiscript
             /// does there is no possible way to recover
             static std::unordered_map<const void*, T> &t() noexcept
             {
-              thread_local std::unordered_map<const void *, T> my_t;
+              static thread_local std::unordered_map<const void *, T> my_t;
```

Comment:
Hmmm, I'm unsure of the best way to represent this.... Should we move the static definition to become a static member of `Thread_Storage`?

Reply:
thread_storage can't have anything else but static members. It's just a normal 'global' variable, but 'global' to thread, not process. Local variables (as it was) are allocated on stack. They aren't global. So you just can't put a local variable into thread_storage, you have to make it global in either way.

Alson in your case it's a singleton (dynamically initialized when first accessed) so in this case it seems natural to make it global by just sticking in 'static'. It's the same as if you just put it on the global scope, but more readable.

Reply:
`thread_local` implies `static` there is no such thing as non-`static` `thread_local` variable in C++11.

This code as compiled for quite some time with VS (~4 years I would think) it's surprising that the most recent VS would break it. But it's a meaningless change regardless, so if it's required in the latest VS than we should just accept it.

Reply:
Actually this raises a completely different question, @yuyaryshev does this mean that the rest of ChaiScript does now compile on VS2017 in C++17 mode and this is the only change necessary?

Reply:
Yep. VS2017 15.8.9 - compiles ok!

Reply:
In fact I even use 'ISO C++ Latest Draft Standard (/std:c++latest)' - so it's ok with the most recent TS's implemented by Microsoft too.